src/core/math/quat.js

```diff
@@ -283,6 +283,41 @@ class Quat {
         return this;
     }
 
+    /**
+     * Returns the result of multiplying the specified quaternions together.
+     *
+     * @param {Quat} lhs - The quaternion used as the first multiplicand of the operation.
+     * @param {Quat} rhs - The quaternion used as the second multiplicand of the operation.
+     * @returns {Quat} The resulting quaternion.
+     * @example
+     * var a = new pc.Quat().setFromEulerAngles(0, 30, 0);
+     * var b = new pc.Quat().setFromEulerAngles(0, 60, 0);
+     *
+     * // r is set to a 90 degree rotation around the Y axis
+     * // In other words, r = a * b
+     * const r = pc.Quat.mul(a, b);
+     *
+     * console.log("The result of the multiplication is: " + r.toString());
+     */
+    static mul(lhs, rhs) {
+        const q1x = lhs.x;
+        const q1y = lhs.y;
+        const q1z = lhs.z;
+        const q1w = lhs.w;
+
+        const q2x = rhs.x;
+        const q2y = rhs.y;
+        const q2z = rhs.z;
+        const q2w = rhs.w;
+
+        return new Quat(
+            q1w * q2x + q1x * q2w + q1y * q2z - q1z * q2y,
+            q1w * q2y + q1y * q2w + q1z * q2x - q1x * q2z,
+            q1w * q2z + q1z * q2w + q1x * q2y - q1y * q2x,
+            q1w * q2w - q1x * q2x - q1y * q2y - q1z * q2z
+        );
```

Comment:
Wondering if adding `Quat#mul2` and using that here would make the API more consistent with vectors/matrices.

Reply:
`Quat#mul2` already exists, same as the previous review I wanted to make it only call `new` instead of `new` + `mul2` but I can change if you prefer